compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -2389,7 +2390,20 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
                         Some((span, open_paren_sp)) => Some((span, true, open_paren_sp)),
                         // If `param` corresponds to `Self`, no usable suggestion span.
                         None if generics.has_self && param.index == 0 => None,
-                        None => Some((self.tcx.def_span(def_id).shrink_to_hi(), false, None)),
+                        None => {
+                            let param = hir_generics
+                                .params
+                                .iter()
+                                .find(|param| param.def_id == def_id)
+                                .unwrap();
+                            let span = if let ParamName::Plain(ident) = param.name {
```

Comment:
I think I'd like us to do something like

```suggestion
                            let span = if let Some(param) = hir_generics
                                .params
                                .iter()
                                .find(|param| param.def_id == def_id)
                                && let ParamName::Plain(ident) = param.name {
```

(needs fmt)